foxpuppet/windows/browser/notifications.py

```diff
@@ -0,0 +1,114 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from abc import ABCMeta
+
+from selenium.webdriver.common.by import By
+
+
+class BaseNotification(object):
+    """Abstract base class for any kind of notification."""
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self, window, root):
+        self.window = window
+        self.root = root
+        self.selenium = self.window.selenium
+
+    @property
+    def label(self):
+        """Provide access to the notification label.
+
+        :returns: The notification label.
+        """
+        with self.selenium.context('chrome'):
+            return self.root.get_attribute('label')
+
+    @property
+    def origin(self):
+        """Provide access to the notification origin.
+
+        :returns: The notification origin.
+        """
+        with self.selenium.context('chrome'):
+            return self.root.get_attribute('origin')
+
+    def close(self):
+        """Close the notification."""
+        with self.selenium.context('chrome'):
+            self.root.find_anonymous_element_by_attribute(
+                'anonid', 'closebutton').click()
+        self.window.wait_for_notification(None)
+
+
+class AddOnInstallBlocked(BaseNotification):
+    """Add-on install blocked notification."""
+
+    def allow(self):
+        """Allow the add-on to be installed"""
+        with self.selenium.context('chrome'):
+            self.root.find_anonymous_element_by_attribute('anonid', 'button') \
+                .find_anonymous_element_by_attribute('anonid', 'button') \
```

Comment:
Having to search with the same arguments twice looks strange. Do we really have a button inside a button?